addons/mail/static/src/js/models/threads/mailbox.js

```diff
@@ -89,31 +67,40 @@ var Mailbox = SearchableThread.extend({
         return this.fetchMessages().then(function (messages) {
             // pick only last message of chatter
             // items = list of objects
-            // { unreadCounter: integer, message: mail.model.Message }
+            // {
+            //    unreadCounter: {integer},
+            //    message: {mail.model.Message},
+            //    messageIDs: {integer[]},
+            // }
             var items = [];
             _.each(messages, function (message) {
                 var unreadCounter = 1;
+                var messageIDs = [message.getID()];
                 var similarItem = _.find(items, function (item) {
                     return self._areMessagesFromSameDocumentThread(item.message, message) ||
                             self._areMessagesFromSameChannel(item.message, message);
                 });
                 if (similarItem) {
                     unreadCounter = similarItem.unreadCounter + 1;
+                    messageIDs = similarItem.messagesIDs.concat(messageIDs);
```

Comment:
This crashes because `similarItem.messagesIDs`. Could you check and fix?

Reply:
Ok, will do.